libraries/AP_PiccoloCAN/AP_PiccoloCAN.cpp

```diff
@@ -691,21 +692,27 @@ bool AP_PiccoloCAN::handle_esc_message(AP_HAL::CANFrame &frame)
     } else if (decodeESC_StatusBPacket(&frame, &esc.voltage, &esc.current, &esc.dutyCycle, &esc.escTemperature, &esc.motorTemperature)) {
         
         AP_ESC_Telem_Backend::TelemetryData t {
-            .temperature_cdeg = int16_t(esc.escTemperature * 100),
             .voltage = float(esc.voltage) * 0.01f,
             .current = float(esc.current) * 0.01f,
+            .motor_temp_cdeg = int16_t(esc.motorTemperature * 100),
```

Comment:
Can't rearrange fields.  It's important our code compiles :-)

Reply:
> We really only want one commit for this change, too :-)

Should I resubmit as a PR with a single commit? Happy to fix all the other points as per your suggestions.